mrmustard/lab_dev/transformations/base.py

```diff
@@ -70,159 +107,134 @@ def inverse(self) -> Transformation:
         self,
         modes_out: tuple[int, ...] = (),
         modes_in: tuple[int, ...] = (),
+        representation: Optional[Bargmann | Fock] = None,
         name: Optional[str] = None,
     ):
         super().__init__(
             modes_out_ket=modes_in,
             modes_in_ket=modes_out,
-            name=name or "Op" + "".join(str(m) for m in modes_in),
-        )
-
-    @classmethod
-    def from_bargmann(
-        cls,
-        modes_out: Sequence[int],
-        modes_in: Sequence[int],
-        triple: tuple[ComplexMatrix, ComplexVector, complex],
-        name: Optional[str] = None,
-    ) -> Operator:
-        A = math.astensor(triple[0])
-        b = math.astensor(triple[1])
-        c = math.astensor(triple[2])
-        shape_check(A, b, len(modes_out) + len(modes_in), "Bargmann")
-        return cls._from_attributes(
-            Bargmann(A, b, c), Wires(set(), set(), set(modes_out), set(modes_in)), name
+            representation=representation,
+            name=name,
         )
 
 
-class Unitary(Operator):
+class Unitary(Operation):
     r"""
     Base class for all unitary transformations.
+    Note the default initializer is in the parent class ``Operation``.
 
     Arguments:
-        name: The name of this transformation.
-        modes: The modes that this transformation acts on.
+        modes_out: The output modes of this Unitary.
+        modes_in: The input modes of this Unitary.
+        representation: The representation of this Unitary.
+        name: The name of this Unitary.
     """
 
-    def __init__(self, name: Optional[str] = None, modes: tuple[int, ...] = ()):
-        super().__init__(
-            modes_in=modes,
-            modes_out=modes,
-            name=name or "U" + "".join(str(m) for m in modes),
-        )
+    short_name = "U"
 
     def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
         r"""
         Contracts ``self`` and ``other`` as it would in a circuit, adding the adjoints when
         they are missing.
 
+        For example ``u >> channel`` is equivalent to ``u.adjoint @ u @ channel`` because the
+        channel requires an input on the bra side as well.
+
         Returns a ``Unitary`` when ``other`` is a ``Unitary``, a ``Channel`` when ``other`` is a
         ``Channel``, and a ``CircuitComponent`` otherwise.
         """
         ret = super().__rshift__(other)
 
         if isinstance(other, Unitary):
-            return Unitary._from_attributes(ret.representation, ret.wires, "")
+            return Unitary._from_attributes(ret.representation, ret.wires)
         elif isinstance(other, Channel):
-            return Channel._from_attributes(ret.representation, ret.wires, "")
+            return Channel._from_attributes(ret.representation, ret.wires)
         return ret
 
-    def __repr__(self) -> str:
-        return super().__repr__().replace("CircuitComponent", "Unitary")
-
     @classmethod
-    def from_bargmann(
+    def from_symplectic(
```

Comment:
Do we have the theory needed to have an analogous `from_XY` for channels? That would be useful for e.g. measurement-based squeezing where we already know the XY map and want to make an MM object for it!

Reply:
Arsalan is on it! 💪